Sources/Swift/SentryOptionsValidator.swift

```diff
@@ -0,0 +1,37 @@
+@objcMembers
+class SentryOptionsValidator: NSObject {
+    @objc
+    static func validate(options: Options) {
+        SentryLog.debug("Validating Sentry SDK options")
+        if !Self.isCacheDirectoryPathValid(path: options.cacheDirectoryPath) {
+            SentryLog.fatal("The configured cache directory path looks invalid, the SDK might not be able to write reports to disk: \(options.cacheDirectoryPath)")
+        }
+    }
+
+    @objc
+    static func isCacheDirectoryPathValid(path: Any) -> Bool {
+        guard let path = path as? String else {
+            return false
+        }
```

Comment:
`m`: It seems odd to me that the `path` is of type `Any`. I would change this to a String and do the casting where it's not a String.

Reply:
The type is `Any` because this method is also used from `SentryOptions.validateOptions:` which is used by Hybrid SDKs to pass type-erased options.

Reply:
What about only doing the type check there?